src/Internal/Payment/PaymentContext.php

```diff
@@ -7,9 +7,92 @@
 
 namespace WCPay\Internal\Payment;
 
+use Automattic\WooCommerce\Blocks\Payments\PaymentMethodTypeInterface;
+use WCPay\Internal\Payment\PaymentMethod\PaymentMethodInterface;
+
 /**
  * A context object, which is shared between payment states.
  */
 class PaymentContext {
+	/**
+	 * Contains all the context's data.
+	 *
+	 * @var array
+	 */
+	private $data = [];
+
+	/**
+	 * Stores an internal value.
+	 * Use this method for changes to allow logging in the future.
+	 *
+	 * @param string $key   Property name.
+	 * @param mixed  $value Value to store.
+	 */
+	private function set( string $key, $value ) {
+		$this->data[ $key ] = $value;
+	}
+
+	/**
+	 * Retrieves an internal value, if any.
+	 *
+	 * @param string $key Key of the property.
+	 * @return mixed|null Either the stored value, or null if not set.
+	 */
+	private function get( string $key ) {
+		return $this->data[ $key ] ?? null;
+	}
+
+	/**
+	 * Stores the order requiring payment.
+	 *
+	 * @param int $order_id ID of the order.
+	 */
+	public function set_order_id( int $order_id ) {
```

Comment:
Nitpick. Do we plan to use PSR12 for function/variable naming as we use PSR4 for class naming (for the sake of autoloader)?

Reply:
I know this feels weird at first, but it is just something to get used to, and that should not be hard, considering we're already using snake case within `includes`.

Also, here we're following the example of Woo core. Random file example: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Orders/TaxesController.php